quasar-core/src/main/java/co/paralleluniverse/fibers/instrument/InstrumentClass.java

```diff
@@ -277,6 +317,28 @@ public void visitEnd() {
         super.visitEnd();
     }
 
+    private boolean compareSuspendableArgs(Type[] expectedMethodTypes, MethodNode mns) {
+        final Type[] candidateMethodTypes = Type.getArgumentTypes(mns.desc);
+        for (int i = 0; i < candidateMethodTypes.length; i++) {
+            if (!Objects.equals(candidateMethodTypes[i], expectedMethodTypes[i + 1])) {
```

Comment:
Do we need to check that `i + 1` is within range for the `expectedMethodTypes` array?

Reply:
Checked size of array before indexing